src/compiler/checker.ts

```diff
@@ -6543,6 +6549,17 @@ namespace ts {
             }
         }
 
+        function signatureHasTypePredicate(signature: Signature): boolean {
+            return signature.resolvedTypePredicate !== undefined;
+        }
+
+        function getTypePredicateOfSignature(signature: Signature): TypePredicate | undefined {

```

Comment:
I've inverted the representation, so `undefined` now represents a lazily-computed type predicate and `noTypePredicate` represents no type predicate existing.